src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -946,6 +935,38 @@ ExecuteTaskListExtended(RowModifyLevel modLevel, List *taskList,
 						TransactionProperties *xactProperties,
 						List *jobIdList,
 						bool localExecutionSupported)
+{
+	bool isUtilityCommand = false;
+	return ExecuteTaskListExtendedInternal(
+		modLevel,
+		taskList,
+		tupleDescriptor,
+		tupleStore,
+		hasReturning,
+		targetPoolSize,
+		xactProperties,
+		jobIdList,
+		localExecutionSupported,
+		isUtilityCommand
+		);
+}
+
+
+/*
+ * ExecuteTaskListExtendedInternal sets up the execution for given task list and
+ * runs it.
+ */
+static uint64
+ExecuteTaskListExtendedInternal(RowModifyLevel modLevel,
+								List *taskList,
+								TupleDesc tupleDescriptor,
+								Tuplestorestate *tupleStore,
+								bool hasReturning,
+								int targetPoolSize,
+								TransactionProperties *xactProperties,
+								List *jobIdList,
+								bool localExecutionSupported,
+								bool isUtilityCommand)
```

Comment:
Is it possible to drop `isUtilityCommand ` flag and infer it in the function via the `taskType`?

(and `localExecutionSupported ` only exists because we don't support VACUUM locally, if you can fix that as well we can remove  it as well :)  )

Reply:
Currently localExecutionSupported exists for `INSERT..SELECT via repartitioning` as well. Once we fix VACUUM and that these calls will be simpler as we can remove all the localExecutionSupported flags.

I think we should be able to infer if a command is a utility command via taskType, I will look into that. That might indeed make things even simpler :)

Reply:
It seems that our task types are pretty generic, so it doesnt seem possible. But we might maybe add some additional information to task struct to understand if it is a utility command

Reply:
Any thoughts on this @marcocitus ?

Reply:
I think using taskType to infer isUtilityCommand (e.g. when it's DDL_TASK) makes sense, although it does bug me a tiny bit that we often end up looking at the first Task to get the properties of the list of tasks (which could be empty). Most of the properties are shared across a list of Tasks.

I suspect we'll never support local execution for VACUUM or CREATE INDEX CONCURRENTLY, so we may need to keep that flag.

Reply:
with a quick look, a DDL_TASK type query may not be always a utility command I guess. It might make sense to add `isUtilityCommand` bool to task maybe.

Reply:
Honestly I am also okay with the current approach after #3747.

Reply:
It became simpler after #3747. Are you guys okay with the current one now @marcocitus, @onderkalaci ?